crates/bevy_gizmos/src/config.rs

```diff
@@ -0,0 +1,161 @@
+//! A module for the [`GizmoConfig<T>`] [`Resource`].
+
+use core::panic;
+use std::{
+    any::Any,
+    ops::{Deref, DerefMut},
+};
+
+use bevy_ecs::{component::Component, system::Resource};
+use bevy_reflect::{Reflect, TypePath};
+use bevy_render::{color::Color, view::RenderLayers};
+use bevy_utils::HashMap;
+
+/// A trait used for custom gizmo configs.
+///
+/// Here you can store additional configuration for you gizmos not covered by [`GizmoConfig`]
+///
+/// Make sure to derive [`Default`] + [`Reflect`], and register in the app using `app.init_gizmo_config::<T>()`
+pub trait CustomGizmoConfig: 'static + Send + Sync + Any + Reflect + TypePath + Default {}
```

Comment:
Hmm, I don't like this too much because it would replace a compiler error when using a wrong type key with a runtime error. Additionally, the number of gizmo configs will be quite low so this won't save many lines of code. But I'm open to following bevy standards.

Reply:
I see what you mean. bevy has been moving away from this type of blanket impls actually, with `Component`, then `Resource`, and `Event` and `Asset`. They all used to have a blanket impl, but now you have to explicitly add a `derive(Foo)`, for the exact reasons you mention.

I think it's fine without the blanket impl, but we should probably have a derive macro for this.

Reply:
I added a derive macro